GetStream/stream-chat-swift
# Do not mark a channel as read if the app is in background

### 🔗 Issue Links

https://github.com/GetStream/stream-chat-swift/issues/2826

### 🎯 Goal

We should not mark channel as read if the app is not in foreground

### 📝 Summary

We are marking as read when ChatChannelConfig.staysConnectedInBackground is true, and the app is backgrounded with the ChatChannelVC "visible".

### 🛠 Implementation

Add a check in `isViewVisible` that makes sure the app is in `.active` state

### 🧪 Manual Testing Notes

0. Set `ChatChannelConfig.staysConnectedInBackground` as `true`
1. Open a channel
2. Background the app
3. Send a message from another device
4. Kill the app
5. Reopen it

Expected result:
- The channel should have the message as unread.

### ☑️ Contributor Checklist


### 🎁 Meme

![](https://media.giphy.com/media/w89ak63KNl0nJl80ig/giphy.gif)
